rpc/metadata.go

```diff
@@ -0,0 +1,42 @@
+package rpc
+
+import (
+	"context"
+	"net/http"
+)
+
+type rawMd struct {
+	headers http.Header
+}
```

Comment:
Any reason for doing it like that instead of `type rawMd http.Header` ?

Reply:
Not sure what `rawMd` is supposed to mean? Maybe name it something more obvious?

Reply:
What I imagine is that there may be other things for flow control/message transmission (not just http headers) in the future. At this time, the rawMd structure can be reused.